src/assets/scripts/client/measurement/MeasureTool.js

```diff
@@ -0,0 +1,363 @@
+import { radians_normalize } from '../math/circle';
+import { bearingToPoint } from '../math/flightMath';
+import { distance2d } from '../math/distance';
+import {
+    heading_to_string,
+    km,
+    nm
+} from '../utilities/unitConverters';
+import MeasureLegModel from './MeasureLegModel';
+import AircraftModel from '../aircraft/AircraftModel';
+import { TIME } from '../constants/globalConstants';
+
+/**
+ * Defines a `MeasureTool`
+ *
+ * The `MeasureTool` is used by the `InputController` to
+ * accept user input and build a list of `MeasureLegModels`
+ * which calculate distances, durations and bearings
+ */
+export default class MeasureTool {
+    /**
+     * @for MeasureLeg
+     * @constructor
+     */
+    constructor() {
+        /**
+         * Indicates whether the tool should be receiving input
+         *
+         * @for `MeasureTool`
+         * @property isMeasuring
+         * @type {boolean}
+         * @default false
+         */
+        this.isMeasuring = false;
+
+        /**
+         * The list of points that make up the path being measured
+         * This can be an object with a `#relativePosition` property,
+         * or a array of numbers that represent a `relativePosition`
+         *
+         * @for `MeasureTool`
+         * @property _points
+         * @type {AircraftModel|FixModel|array<number>}
+         * @default []
+         * @private
+         */
+        this._points = [];
+    }
+
+    /** Gets the aircraft at the beginning of the path
+     *
+     * @for MeasureTool
+     * @property aircraft
+     * @type {AircraftModel|null}
+     */
+    get aircraft() {
+        if (this.hasStarted && this._points[0] instanceof AircraftModel) {
```

Comment:
please reverse this logic to return `null` early when we don't have what we need